pages/custom-query-modules/rust/rust-example.mdx

```diff
@@ -0,0 +1,214 @@
+```
+
+### Enter into memgraph container shell
+
+After you have started the Memgraph instance, you need to enter the shell. You can do this by running the following command:
+
+```bash
+docker exec -it -u root mage bash
+```
+
+This will start bash inside the container as the root user. Since we will need to install certain dependencies, we need to be the root user. Of course, if you are running a native Linux machine, you can skip this step.
+
+### Install Cargo for Rust
+
+Query modules in Rust are compiled inside the Memgraph image using `cargo`. Cargo has a large overhead on the Docker image and it is not
+installed by default. In order to install cargo and add it to the `PATH` environment variable, we will use the following commands:
+
+```bash
+curl https://sh.rustup.rs -sSf | sh -s -- -y
+export PATH="/root/.cargo/bin:${PATH}"
+```
+
+### Create the file structure for the query module
+
+Now that all of the required dependencies are installed, you can create a directory for your query module. In the rust directory (`/mage/rust`), you
+can copy the `rsmgp-example` boilerplate file structure and modify the following information:
+
+In the `Cargo.toml` file, apply the changes:
+```toml
+[package]
+name = "your-package-name"
+
+[lib]
+name = "your-query-module-top-level-name"
+```
+
+The complete `Cargo.toml` file should look like this:
+```toml
+[package]
+name = "rsmgp-my-query-module-package"
+version = "0.1.0"
+authors = ["Josip Mrden <josip.mrden@memgraph.io>"]
+edition = "2018"
+
+[dependencies]
+c_str_macro = "1.0.2"
+rsmgp-sys = { path = "../rsmgp-sys" }
+
+[lib]
+name = "my_rust_query_module"
+crate-type = ["cdylib"]
+```
+
+### Develop your Rust query module
+
+Inside the `lib.rs` file of the query module file structure, you can add your specific 
+module using the [Rust API](/custom-query-modules/rust/rust-api.md).
+
+An example query module file would look like this
+
+```rs
+use c_str_macro::c_str;
+use rsmgp_sys::list::*;
+use rsmgp_sys::memgraph::*;
+use rsmgp_sys::mgp::*;
+use rsmgp_sys::property::*;
+use rsmgp_sys::result::*;
+use rsmgp_sys::rsmgp::*;
+use rsmgp_sys::value::*;
+use rsmgp_sys::{close_module, define_optional_type, define_procedure, define_type, init_module};
+use std::ffi::CString;
+use std::os::raw::c_int;
+use std::panic;
+
+init_module!(|memgraph: &Memgraph| -> Result<()> {
+    memgraph.add_read_procedure(
+        basic,
+        c_str!("basic"),
+        &[define_type!("input_string", Type::String)],
+        &[define_optional_type!(
+            "optional_input_int",
+            &MgpValue::make_int(0, &memgraph)?,
+            Type::Int
+        )],
+        &[
+            define_type!("output_string", Type::String),
+            define_type!("output_int", Type::Int),
+        ],
+    )?;
+
+    Ok(())
+});
+
+define_procedure!(basic, |memgraph: &Memgraph| -> Result<()> {
+    // This procedure just forwards the input parameters as procedure results.
```

Comment:
This sounds a lot like "echo". Changing the name to something more meaningful than "basic" is a good idea.